payment_methods/SquareOnsite/forms/SettingsForm.php

```diff
@@ -242,4 +184,77 @@ public function validateConnection(EE_Payment_Method $pmInstance)
             )
         );
     }
+
+
+    /**
+     * Checks if the OAuth credentials are still healthy and we are authorized.
+     *
+     * @param EE_Payment_Method $pmInstance
+     * @return array ['healthy' => true] | ['error' => ['message' => 'the_message', 'code' => 'the_code']]
+     */
+    public function oauthHealthCheck(EE_Payment_Method $pmInstance)
+    {
+        // Request a list of locations.
+        $locations = EED_SquareOnsiteOAuth::getMerchantLocations($pmInstance);
+        if (is_array($locations) && isset($locations['error'])) {
+            switch ($locations['error']['code']) {
+                case 'ACCESS_TOKEN_EXPIRED':
+                case 'ACCESS_TOKEN_REVOKED':
+                case 'INSUFFICIENT_SCOPES':
+                    // We have an error. Put it under NOT_AUTHORIZED category for easy identification.
+                    $locations['error']['code'] = 'NOT_AUTHORIZED';
+                    return $locations;
+                default:
+                    return $locations;
+            }
+        }
+        return ['healthy' => true];
+    }
+
+
+    /**
+     * Resets all that OAuth related settings.
+     *
+     * @param EE_Payment_Method $pmInstance
+     * @return boolean
+     * @throws ReflectionException
+     */
+    public function resetOauthSettings(EE_Payment_Method $pmInstance)
+    {
+        try {
+            $pmInstance->delete_extra_meta(Domain::META_KEY_APPLICATION_ID);
+            $pmInstance->delete_extra_meta(Domain::META_KEY_ACCESS_TOKEN);
+            $pmInstance->delete_extra_meta(Domain::META_KEY_LOCATION_ID);
+            $pmInstance->update_extra_meta(
+                Domain::META_KEY_SQUARE_DATA,
+                [
+                    Domain::META_KEY_USING_OAUTH => false,
+                ]
+            );
+        } catch (EE_Error $e) {
+            return false;
+        }
+        return true;
+    }
+
+
+    /**
+     * This simply adds the form validation error.
+     *
+     * @param string $errorMessage has to have two placeholders for the bold text.
+     * @param string $errorName
+     * @return void
+     */
+    protected function doValidationError(string $errorMessage, string $errorName)
+    {
+        $this->add_validation_error(
+            sprintf(
+                // translators: %1$s: opening strong html tag. $2$s: closing strong html tag.
+                esc_html__($errorMessage, 'event_espresso'),
```

Comment:
this doesn't work unfortunately because the strings are not parsed by the PHP engine, therefore the value of `$errorMessage` is not known when the POT file is generated.

```suggestion
                $errorMessage,
```

Reply:
A result of "more refactoring, more.."